File: city-tours-admin/src/services/auth/AuthApi.js
import axiosClient from "../AxiosClient";
import { API_ROUTE } from "../../utils/Constants";

const authApi = {
  register: async (data) => {
    const url = API_ROUTE.REGISTER;
    try {
      const response = await axiosClient.post(url, data);
      console.log("response Api: ", response);
      return response;
    } catch (error) {
      console.log("response Api error: ", error);
      return { error };
    }
  },

  login: async (data) => {
    const url = API_ROUTE.LOGIN;
    try {
      const response = await axiosClient.post(url, data);
      console.log("response Api: ", response);
      return response;
    } catch (error) {
      console.log("response Api error: ", error);
      return { error };
    }
  },

  logout: async () => {
    const url = API_ROUTE.LOGOUT;
    try {
      const response = await axiosClient.post(url);
      console.log("response Api: ", response);
      return response;
    } catch (error) {
      console.log("response Api error: ", error);
      return { error };
    }
  },
};

export default authApi;

File: city-tours-mobile/lib/src/services/auth_api.dart
import 'dart:convert';
import 'package:flutter/src/widgets/framework.dart';
import 'package:flutter_application_1/src/models/auth_model.dart';
import 'package:flutter_application_1/src/state/auth_provider.dart';
import 'package:flutter_application_1/src/utils/constants.dart';
import 'package:http/http.dart' as http;
import 'package:provider/provider.dart';

class AuthApi {
  // Register
  Future<dynamic> register(
      String username, String email, String password) async {
    const url = URI.REGISTER_MOBILE;
    final Map<String, String> headers = {
      'Content-Type': 'application/json; charset=UTF-8',
    };
    final Map<String, dynamic> body = {
      'username': username,
      'email': email,
      'password': password,
      'roles': ['ROLE_CUSTOMER']
    };

    final response = await http.post(
      Uri.parse(url),
      headers: headers,
      body: jsonEncode(body),
    );

    if (response.statusCode == 201) {
      final responseJson = jsonDecode(response.body)['data'];
      final auth = Auth.fromJson(responseJson);
      return auth;
    } else {
      final errorMessage = jsonDecode(response.body)['message'];
      throw errorMessage;
    }
  }

  // Login
  Future<dynamic> login(
      BuildContext context, String username, String password) async {
    const url = URI.LOGIN_MOBILE;
    final Map<String, String> headers = {
      'Content-Type': 'application/json; charset=UTF-8',
    };
    final Map<String, dynamic> body = {
      'username': username,
      'password': password,
    };

    final response = await http.post(
      Uri.parse(url),
      headers: headers,
      body: jsonEncode(body),
    );

    if (response.statusCode == 200) {
      final responseJson = jsonDecode(response.body);
      final token = responseJson['data']['token'];

      final authProvider = Provider.of<AuthProvider>(context, listen: false);
      authProvider.setToken(token);
      authProvider.setUsername(username);
    } else {
      final errorMessage = jsonDecode(response.body)['message'];
      throw errorMessage;
    }
  }
}
